src/classes/aws-utils.ts

```diff
@@ -0,0 +1,79 @@
+import fs from "fs";
+import { ClusterConfig, OperatingSystems } from "../interfaces/types";
+import OsUtils from "./os-utils";
+import { PathUtils } from "./path-utils";
+import YamlUtils from "./yaml-utils";
+
+const { HOME } = process.env;
+let configPath = `${HOME}/.aws/config`;
+
+/**
+ * Provides utilities for setting up AWS credentials, clusters and Kube configuration for AWS clusters
+ */
+export class AWSUtils {
+
+  /**
+   * Configures AWS CLI credentials for eksctl & aws cli usage
+   * 
+   * @param projName Project name that is also the cluster name
+   * @param access AWS access key ID
+   * @param secret AWS secret access key
+   * @returns {string[]} array of commands to run config .sh in the main thread
+   */
+  public static configAWS = async (projName: string, access? : string, secret?: string): Promise<string[]> => {
+    try {
+      configPath = await PathUtils.fixPath(configPath);
+      const os: string = await OsUtils.getOS();
+      return (os == OperatingSystems.LINUX || os == OperatingSystems.MAC) ?
+      ["chmod +x aws-configure.sh", `./aws-configure.sh -n ${projName} ${access} ${secret}`] 
+      : [];
+
+    } catch (err) {
+      return Promise.reject(`Error when trying to config AWS: ${err}`);
+    }
+  }
+
+  /**
+   * Creates cluster .yaml that is used to create the cluster itself
+   * 
+   * @param name Project name / cluster name
+   * @param args User input arguments
+   * @param repoPath Repository path where to init cluster.yaml
+   */
+  public static createCluster = async (name: string, args: ClusterConfig, repoPath: string): Promise<void> => {
+    if (!args.vpcIP) {
+      return Promise.reject("No VPC CIDR IP has been set.");
+
+    } else {
+    if (!args.desiredCapacity) {
+      args.desiredCapacity = 1;
+    }
+    if (!args.minSize) {
+      args.minSize = 1;
+    }
+    if (!args.maxSize) {
+      args.maxSize = 2;
+    }
+    if (!args.volumeSize) {
+      args.volumeSize = 20;
+    }
+    if (!args.clusterLabel) {
+      args.clusterLabel = "cluster-ops";
+    }
+    if (args.ngLabel) {
+      args.ngLabel = "frontend-workloads";
+    }
```

Comment:
These would proably look better in ternary operator format (var ? if true : if false;)

Reply:
solution var ? var = var : var = somethingElse results into error: Operator cannot be assigned to itself
solution var ? var : var = somethingElse results into bug

Reply:
so this wouldnt work?
(!args?.volumesize) ? args.volumesize : 20;

Reply:
Results into a bug